### PR TITLE
Run documentation build and doctest in GitHub Actions.

### PR DESCRIPTION
**Pull request by crcrpar**
We want to migrate all the required checks to github actions as we saw some weird connection problems with Circle CI (at least twice).

It may or may not be better to use [the official images of readthedocs](https://hub.docker.com/r/readthedocs/build).

### CODE REVIEW

File: .github/workflows/sphinx-build.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    - name: Install Dependencies
+      run: |
+        python -m pip install -U pip
+        pip install --progress-bar off -U .[document]
+
+    - name: Build Document
+      run: |
+        cd docs
+        make html -j
```
Comment (toshihikoyanase) [inline .github/workflows/sphinx-build.yml:30]: Why did you add `-j` option?
It was ignored in the `Makefile` when I dumped the following command:

https://github.com/optuna/optuna/blob/dc032afdda0e0248bb70831ee47ca10aeca3bd28/docs/Makefile#L20

The command dump:

```console
$ make html -j
echo @sphinx-build -M html "source" "build" -W --keep-going 
@sphinx-build -M html source build -W --keep-going
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR as always.

> It may or may not be better to use the official images of readthedocs.

I prefer the official image since we can reduce environment-dependent issues. On the other hand, this [document](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on) says that we have only five supported runners, and I guess we cannot use the custom images.

File: .github/workflows/sphinx-build.yml (hunk 1 above)
Comment (crcrpar) [reply .github/workflows/sphinx-build.yml]: It's just my habit. Removed.

**Comment by crcrpar** [comment]
We can set a container with `container` argument. Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer

Though, after some investigation, I found that I can't make that happen at the moment because
- `actions/checkout` requires git 2.18 or higher
- the version of `readthedocs/build` images' git is 2.17
- sudo / su not available in `readthedocs/build`

File: .github/workflows/sphinx-build.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - uses: actions/checkout@v2
+
+    - uses: actions/setup-python@v2
+      with:
+        python_version: 3.8
```
Comment (HideakiImamura) [inline .github/workflows/sphinx-build.yml:20]: A nit. 
```suggestion
        python-version: 3.8
```

File: .github/workflows/sphinx-build.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
```
Comment (HideakiImamura) [inline .github/workflows/sphinx-build.yml:48]: ditto.
```suggestion
        python-version: 3.8
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR! The added workflow almost LGTM. I have minor comments. Could you take a look?

**Review by HideakiImamura — APPROVED** [review]
Thanks for the swift action! LGTM!

**Review by toshihikoyanase — APPROVED** [review]
LGTM. Thank you!
